test/jsonValidationTest.js

```diff
@@ -85,8 +97,46 @@ describe('Validating Workbooks...', () => {
             done();
         }, true, workbookPath);
     });
+
+    it('Verifying workbook category or settings json exists', function (done) {
+        browseDirectory(workbookPath, (error, results) => {
+            if (error) throw error;
+            results.filter(file => file.substr(-9) === '.workbook')
+                .forEach(folder => {
+                    validateJsonExistForWorkbook(workbookPath, results, folder.substr(workbookPath.length+1))
+                });
+
+            done();
+        }, true, workbookPath);
+    });
 });
 
+function validateJsonExistForWorkbook(rootPath, results, file) {
+    let paths = getProgressivePaths(rootPath, file);
```

Comment:
the way this is written, doesn't this mean that if you have 10 workbooks in a folder, the parent folders get checked 10 times, since getProgressivePaths gives you every intermediate path?  probably fine but seems like it's a lot of extra checks as the tree gets bigger?

Reply:
good catch. I did notice that as well, however, the test runs in 41ms for the entire tree so it's fast. usually the leaf if 3 folder deep and it was worth doing the extra check to match the entire folder name